### ISSUE TITLE
"Your proxy requires authentication." behind Kerberos AD authenticated proxy

### ISSUE THREAD

**Issue by samthebest** [sbt/sbt#1781]
## steps

So I've tried kinit and using the following flags

```
export SBT_OPTS="-Dhttp.proxyHost=... -Dhttp.proxyPort=8080 -Dhttps.proxyHost=... -Dhttps.proxyPort=8080"
```

Here I don't include the user and password, and I get

```
[error] Unable to find credentials for ...
```

If I include the username and password...
## problems

If I include the username and password I get

```
[warn] Your proxy requires authentication.
```

Also I shouldn't have to save my password into environment variables that is a major security problem.
## notes

I can get curl to work behind my proxy, and a few other tools, also my browser works fine. This must be some kind of bug in SBT.

**Comment by ionFreeman** [comment]
@samthebest? What happened with this?

**Comment by OpsRJ** [comment]
Same here. I'm not able to use sbt behind a proxy with authentication.

**Comment by ionFreeman** [comment]
@OpsRJ I think I had this working with a .credentials file http://www.scala-sbt.org/0.13/docs/Publishing.html
I still had to keep my password in plain text. I got around this by using JFrog Artifactory, which generates an API key you can use in lieu of a password.

**Comment by EdgarLemaire** [comment]
Hello ! I encountered this problem while using SBT and found a way to solve it. Since this topic is more than 1 year old, I suppose that you overcame this problem or gave up on it, but I think this post could help somebody else.
First of all, it is important to note that I am using Ubuntu 16.04 (Xenial), thus this solution might not be helping for other OS users.
My problem was : 
-I am using SBT behind a corporate proxy which requires authentication
-When using SBT, I encountered "Your Proxy Requires Authentication" warning message, resulting in an Error (something like "module not found", as SBT could not access the internet to look for specific packages etc...).

The solution I found : 
-I installed a Squid Proxy on my machine.
-I configured the Squid Proxy to access my corporate proxy.
-Instead of configuring SBT for my corporate proxy, I configured SBT to access my local Squid Proxy, by exporting JAVA_OPTS environment variable as follows : 

> export JAVA_OPTS="-Dhttps.proxyHost=localhost -Dhttps.proxyPort=3128 -Dhttp.proxyHost=localhost -Dhttp.proxyPort=3128" 

Localhost refers to Squid Proxy host name, and 3128 to Squid Proxy port. Copying and pasting this command in your bashrc file without changing anything should work just fine.

Some explanations : 
I think that SBT is not able to handle proxies when they require authentication. Thus, I by-passed the problem by creating an intermediate between SBT and my corporate proxy, namely the local Squid Proxy. SBT transmits request to the Squid Proxy which does not require authentication, then the Squid Proxy transfer the request to my Corporate Proxy, performing authentication independently from SBT.

Hope this will help some people. I apologize for my poor skills in computing, which might results in ugly approximations in this present post.

Regards,

Edgar
